From 6c8c55f54609a76f77cfa0619046ef764b68994f Mon Sep 17 00:00:00 2001
From: Li Huafei <lihuafei1@huawei.com>
Date: Wed, 28 Dec 2022 11:26:52 +0800
Subject: [PATCH 0883/1398] [Huawei] watchdog/corelockup: Support corelockup on
 X86_64

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188185

--------------------------------

The corelockup implementation only relies on softlockup and hardlockup
and has no architecture-related code of its own. Currently, only ARM64
and X86_64 are verified, and X86_64 is added to enable it.

Also, corelockup only supports multicore systems, making it dependent on
CONFIG_SMP.

Signed-off-by: Li Huafei <lihuafei1@huawei.com>
---
 lib/Kconfig.debug | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 10e425c30486..de42111fc4a0 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1023,7 +1023,8 @@ config HARDLOCKUP_DETECTOR
 config CORELOCKUP_DETECTOR
 	bool "Detect Core Lockups"
 	depends on HARDLOCKUP_DETECTOR && SOFTLOCKUP_DETECTOR
-	depends on ARM64
+	depends on ARM64 || X86_64
+	depends on SMP
 	default n
 	help
 	  Corelockups is used to check whether cpu core hungup or not.
-- 
2.17.1

